ostree-ext/src/json.rs

```diff
@@ -0,0 +1,46 @@
+use anyhow::Result;
+use serde::Serialize;
+use std::io;
+
+/// Ensure reproducible JSON output by ordering map keys
+pub(crate) trait JsonOrderedSerialize {
+    fn to_json_ordered_string(&self) -> Result<String>;
+    fn to_json_ordered_writer<W>(&self, writer: W) -> Result<()>
+    where
+        W: io::Write;
+}
+
+impl<S> JsonOrderedSerialize for S
+where
+    S: Serialize,
+{
+    fn to_json_ordered_string(&self) -> Result<String> {
+        let val = serde_json::to_value(self)?;
```

Comment:
Not that the JSON we serialize here is large, but this is inefficient. It's also not really clear to me why this solves the problem...actually it looks like it only does if a not-on-by-default feature is passed https://github.com/serde-rs/json/blob/c1826ebcccb1a520389c6b78ad3da15db279220d/Cargo.toml#L61

I think we really just need to fix the olpc json formatter crate. In order to not block on upstream I'd tentatively be OK forking it into ocidir or so if that helps.

Reply:
> I think we really just need to fix the olpc json formatter crate. In order to not block on upstream I'd tentatively be OK forking it into ocidir or so if that helps.

https://github.com/containers/ocidir-rs/pull/39